infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -121,6 +120,8 @@ public class CorfuReplicationDiscoveryService implements CorfuReplicationDiscove
     /**
      * Current node's id
      */
+    @Getter
+    @VisibleForTesting
```

Comment:
there is a TODO to move the logic of getting the nodeId into the plugin.
Also, we do use DefaultClusterManager.getDefaultNodeId to get the localNodeId for tests.

Also, since this is inline with @pankti-m  previous comment https://github.com/CorfuDB/CorfuDB/pull/3510#discussion_r1096429428, I guess its better to pass the endpoint in the DefaultClusterManager's constructor..?
please let me know your thoughts.

Reply:
Actually no, I read the classes wrong (mixed `DefaultClusterManager` and `DefaultClusterConfig`). 
We cannot pass the endpoint to DefaultClusterManager for our testing, that would change the way we initiate the plugin.

Reply:
Sorry I didn't follow, which plugin are you referring to? 

I thought we set `localNodeId` by parsing plugin config's "local_node_id_path" property, but actually don't have a "default local node id file" in any of our properties file for testing (like `nettyConfig.properties`). So `nodeIdFilePath` has been always null in our test and read from "getDefaultNodeId()" method in `DefaultClusterConfig`.

So I was thinking anyway it's set by external plugin, maybe we can get it directly from outside instead of exposing this field from discovery service only for testing/setup purpose.

Reply:
I believe what Chris is exposing is the issue, the plugin sets the file path to the local node id and then the logic to parse that file is encoded inn the discovery service, but the parsing logic is very specific to the format of that file (which is not expected to be generic/standard for different clients). So the node Id howled come from the plugin and not only the file path. Then, there would be no exception from the default implementations used for testing.

Reply:
discussed offline and decision is to address the TODO in a different PR